detectable/src/main/java/com/synopsys/integration/detectable/detectables/clang/packagemanager/resolver/DpkgPackageManagerResolver.java

```diff
@@ -50,22 +52,33 @@ public DpkgPackageManagerResolver(DpkgPkgDetailsResolver versionResolver) {
         String[] packageLines = ownershipQueryOutput.split("\n");
         for (String packageLine : packageLines) {
             if (!valid(packageLine)) {
-                logger.trace(String.format("Skipping file ownership query output line: %s", packageLine));
+                logger.trace("Skipping file ownership query output line: {}", packageLine);
                 continue;
             }
             String[] queryPackageOutputParts = packageLine.split("\\s+");
             String[] packageNameArchParts = queryPackageOutputParts[0].split(":");
             String packageName = packageNameArchParts[0].trim();
-            logger.debug(String.format("File ownership query results: package name: %s", packageName));
-            Optional<PackageDetails> pkg = versionResolver.resolvePackageDetails(currentPackageManager, executableRunner, workingDirectory, packageName);
+            String packageArch = parseArchitectureIfPresent(packageNameArchParts);
+            logger.debug("File ownership query results: package name: {}, arch: {}", packageName, packageArch);
+            Optional<PackageDetails> pkg = versionResolver.resolvePackageDetails(currentPackageManager, executableRunner, workingDirectory, packageName, packageArch);
             if (pkg.isPresent()) {
-                logger.debug(String.format("Adding package: %s", pkg.get()));
+                logger.debug("Adding package: {}", pkg.get());
                 packageDetailsList.add(pkg.get());
             }
         }
         return packageDetailsList;
     }
 
+    @Nullable
+    private String parseArchitectureIfPresent(String[] packageNameArchParts) {
+        if (packageNameArchParts.length > 1) {
```

Comment:
I'm not a big fan that we pass this array in here and access it in this way because in order to understand this method I have to go find who is calling it and how that array was built and already accessed. 

I think I'd like it if 58-61 moved into this method and it returns an object of Name/Architecture. At least that way all the handling of this array is together.